google/cloud/aiplatform/jobs.py

```diff
@@ -1913,3 +1918,505 @@ def run(
+                on the same endpoint passed in the argument. If `objective_configs` is a dictionary,
+                then this parameter is ignored. If `objective_configs` is an instance of
+                `model_monitoring.EndpointObjectiveConfig` and `deployed_model_ids` is a non-empty
+                list of valid IDs, then the same objective config will apply to all models in this list.
+
+        Returns:
+            A List of ModelDeploymentMonitoringObjectiveConfig objects.
+
+        Raises:
+            ValueError, when the model IDs given are invalid.
+            RuntimeError, when XAI is enabled on a model that doesn't have XAI parameters configured.
+        """
+        all_models = []
+        xai_enabled = []
+        for model in endpoint.list_models():
+            all_models.append(model.id)
+            if str(model.explanation_spec.parameters) != "":
+                xai_enabled.append(model.id)
+
+        all_configs = []
+
+        ## when same objective config is applied to SOME or ALL models
+        if deployed_model_ids is not None:
+            if not all(model in all_models for model in deployed_model_ids):
+                error_string = (
+                    "Invalid model ID. The model ID must be one of ["
+                    + ",".join(all_models)
+                    + "]. Note that deployed model IDs are different from the uploaded model's ID"
+                )
+                raise ValueError(error_string)
+            else:
+                all_models = deployed_model_ids
+
+        if isinstance(objective_configs, model_monitoring.EndpointObjectiveConfig):
+            for model in all_models:
+                if (
+                    model not in xai_enabled
+                    and objective_configs.explanation_config is not None
+                ):
+                    raise RuntimeError(
+                        "Invalid config for model ID %s. `explanation_config` should only be enabled if the model has `explanation_spec populated"
+                        % model
+                    )
+                all_configs.append(
+                    gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringObjectiveConfig(
+                        deployed_model_id=model,
+                        objective_config=objective_configs.as_proto(),
+                    )
+                )
+
+        ## when different objective configs are applied to EACH model
+        else:
+            if not all(model in all_models for model in objective_configs.keys()):
+                error_string = (
+                    "Invalid model ID. The model ID must be one of ["
+                    + ",".join(all_models)
+                    + "]. Note that deployed model IDs are different from the uploaded model's ID"
+                )
+                raise ValueError(error_string)
+            for (key, value) in objective_configs.items():
```

Comment:
Use more descriptive variable names, such as 
```
for deployed_model_id, objective_config in objective_configs.items(): ...
```

so that the next contributor does not need to look up what `objective_configs` contains.

Reply:
i have updated it in the latest commit